src/lib.rs

```diff
@@ -1,88 +1,298 @@
-            subtitle: *const NSString,
-            message: *const NSString,
-            sound: *const NSString,
-            deliveryDate: f64,
-        ) -> bool;
         pub fn sendNotification(
             title: *const NSString,
             subtitle: *const NSString,
             message: *const NSString,
-            sound: *const NSString,
-        ) -> bool;
+            options: *const NSDictionary<NSString, NSString>,
+        ) -> Id<NSDictionary<NSString, NSString>>;
         pub fn setApplication(newbundleIdentifier: *const NSString) -> bool;
         pub fn getBundleIdentifier(appName: *const NSString) -> *const NSString;
     }
 }
 
-/// Schedules a new notification in the NotificationCenter
-///
-/// Returns a `NotificationError` if a notification could not be scheduled
-/// or is scheduled in the past
-///
-/// # Example:
-///
-/// ```ignore
-/// extern crate chrono;
-/// # use mac_notification_sys::*;
-/// use chrono::offset::*;
-///
-/// // schedule a notification in 5 seconds
-/// let _ = schedule_notification("Title", &None, "This is the body", &Some("Ping"),
-///                               Utc::now().timestamp() as f64 + 5.).unwrap();
-/// ```
-pub fn schedule_notification(
-    title: &str,
-    subtitle: &Option<&str>,
-    message: &str,
-    sound: &Option<&str>,
-    delivery_date: f64,
-) -> NotificationResult<()> {
-    ensure!(
-        delivery_date >= Utc::now().timestamp() as f64,
-        NotificationError::ScheduleInThePast
-    );
-
-    let use_sound = match sound {
-        Some(sound) if check_sound(sound) => sound,
-        _ => "_mute",
-    };
-    unsafe {
-        ensure!(
-            sys::scheduleNotification(
-                NSString::from_str(title).deref(),
-                NSString::from_str(subtitle.unwrap_or("")).deref(),
-                NSString::from_str(message).deref(),
-                NSString::from_str(use_sound).deref(),
-                delivery_date,
+/// Possible actions accessible through the main button of the notification
+pub enum MainButton<'a> {
+    /// Display a single action with the given name
+    ///
+    /// # Example:
+    ///
+    /// ```no_run
+    /// # use mac_notification_sys::*;
+    /// let _ = MainButton::SingleAction("Action name");
+    /// ```
+    SingleAction(&'a str),
+    /// Display a dropdown with the given title, with a list of actions with given names
+    ///
+    /// # Example:
+    ///
+    /// ```no_run
+    /// # use mac_notification_sys::*;
+    /// let _ = MainButton::DropdownActions("Dropdown name", &["Action 1", "Action 2"]);
+    /// ```
+    DropdownActions(&'a str, &'a [&'a str]),
+    /// Display a text input field with the given placeholder
+    ///
+    /// # Example:
+    ///
+    /// ```no_run
+    /// # use mac_notification_sys::*;
+    /// let _ = MainButton::Response("Enter some text...");
+    /// ```
+    Response(&'a str),
+}
+
+/// Options to further customize the notification
+pub struct NotificationOptions<'a> {
```

Comment:
Maybe we could Make something like an NotificationOptionsBuilder? 
So if you just want a sound you could use `NotificationOptions::new().sound("Blow").build()` and so `::new()` returns the builder which always returns it self modified and a NotificationOptions if you call `build()`